CodeTracker/CodeTrackerUnitTest/CodingTrackerInputTest.cs

```diff
@@ -0,0 +1,102 @@
+using CodeTracker;
+using Spectre.Console.Testing;
+namespace CodeTrackerUnitTest;
+
+[TestClass]
+public class CodingTrackerInputTest
+{
+    private readonly TestConsole _console;
+    private readonly Input _input;
+    public CodingTrackerInputTest()
+    {
+        _console = new TestConsole();
+        _input = new Input(_console);
+    }
+    [TestMethod]
+    [DataRow(" ")]
+    [DataRow("description")]
+    public void TestingDescriptionInput(string input)
+    {
+        // Arrange
+        if (!string.IsNullOrWhiteSpace(input))
+        {
+            _console.Input.PushTextWithEnter(input);  // Valid input case
+            var result = _input.GetDescription();
+            Assert.AreEqual(input, result);  // Verify the description matches the input
+        }
+        else
+        {
+            _console.Input.PushTextWithEnter(input);  // Invalid input case
+
+            // Act & Assert: Expect an exception due to invalid input
+            Assert.ThrowsException<InvalidOperationException>(() => _input.GetDescription());
+        }
+    }
+
+    [TestMethod]
+    [DataRow("test")]
+    [DataRow("")]
+    [DataRow("22-10-2023")]
+    public void TestDateInput(string input)
+    {
+        var reader = new StringReader(input);
+        Console.SetIn(reader);
+        DateTime result = _input.GetDay();
+        if (!input.Any(char.IsDigit))
+        {
+            Assert.AreEqual(DateTime.Today,result);
+            return;
+        }
+        Assert.AreEqual(string.IsNullOrEmpty(input) ? DateTime.Today : new DateTime(2023,10,22), result);
+    }
+    [TestMethod]
+    [DataRow("20:30")]
+    public void TestStartTimeInput(string input)
```

Comment:
🟠 This test is straightforward but only checks a single input format. It does not account for invalid time formats or edge cases (e.g., `25:00`).

💡 Add tests for invalid time strings and edge cases (e.g., empty string, non-numeric input).

🔧 Consider separating the date and time validation logic if they are mixed in the implementation.